### PR TITLE
Add "Verifying your contracts" guide

### PR DESCRIPTION
**Pull request by fvictorio**

### CODE REVIEW

File: docs/src/content/hardhat-runner/docs/guides/verifying.md [hunk 1]
```diff
@@ -0,0 +1,155 @@
+# Verifying your contracts
+
+Once your contract is ready, the next step is to deploy it to a live network and verify its source code.
+
+Verifying a contract means making public its source code and the compiler settings you used, which allows anyone to compile it and compare the generated bytecode with the one that is deployed on-chain. Doing this is extremely important in an open platform like Ethereum.
```
Comment (feuGeneA) [inline docs/src/content/hardhat-runner/docs/guides/verifying.md:5]:
```suggestion
Verifying a contract means making its source code public, along with the compiler settings you used, which allows anyone to compile it and compare the generated bytecode with the one that is deployed on-chain. Doing this is extremely important in an open platform like Ethereum.
```

File: docs/src/content/hardhat-runner/docs/guides/verifying.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+In this guide we'll explain how to do this in the [Etherscan](https://etherscan.io/) explorer, but there are other ways to verify a contract like [Sourcify](https://sourcify.dev/).
```
Comment (feuGeneA) [inline docs/src/content/hardhat-runner/docs/guides/verifying.md:7]:
```suggestion
In this guide we'll explain how to do this in the [Etherscan](https://etherscan.io/) explorer, but there are other ways to verify a contract, for example with [Sourcify](https://sourcify.dev/).
```

File: docs/src/content/hardhat-runner/docs/guides/verifying.md [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+## Getting an API key from Etherscan
+
+The first thing you need to is an API key from Etherscan. To get one, go to [their site](https://etherscan.io/login), sign in (or create an account if you don't have one) and open the "API Keys" tab. Then click the "Add" button and give a name (like "Hardhat") to the API key you are creating. After that you'll see the newly created key in the list.
```
Comment (feuGeneA) [inline docs/src/content/hardhat-runner/docs/guides/verifying.md:11]:
```suggestion
The first thing you need is an API key from Etherscan. To get one, go to [their site](https://etherscan.io/login), sign in (or create an account if you don't have one) and open the "API Keys" tab. Then click the "Add" button and give a name (like "Hardhat") to the API key you are creating. After that you'll see the newly created key in the list.
```

File: docs/src/content/hardhat-runner/docs/guides/verifying.md [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+::::tabsgroup{options=TypeScript,JavaScript}
+
+:::tab{value=TypeScript}
+
+```ts
+// Go to https://www.alchemyapi.io, sign up, create
+// a new App in its dashboard, and replace "KEY" with its key
+const ALCHEMY_API_KEY = "KEY";
+
+// Replace this private key with your Goerli account private key
```
Comment (feuGeneA) [inline docs/src/content/hardhat-runner/docs/guides/verifying.md:58]:
```suggestion
// Replace this private key with your Goerli account private key.
```

File: docs/src/content/hardhat-runner/docs/guides/verifying.md [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+// To export your private key from Metamask, open Metamask and
+// go to Account Details > Export Private Key
```
Comment (feuGeneA) [inline docs/src/content/hardhat-runner/docs/guides/verifying.md:60]:
```suggestion
// go to Account Details > Export Private Key.
```

**Review by feuGeneA — APPROVED** [review]

**Comment by fvictorio** [comment]
Thank you @feuGeneA. I'll keep this one open one day more in case @alcuadrado wants to review it and then I'll merge it.

**Pull request merged**